sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -181,11 +195,111 @@ public SeekableByteChannel newByteChannel(Path path, Set<? extends OpenOption> s
     }
 
     /**
+     * The existence of a directory in the {@code AzureFileSystem} is defined on two levels. Weak existence is defined
+     * by the presence of a non-zero number of blobs prefixed with the directory's path. This is also known as a virtual
+     * directory and is in order to accommodate containers that were pre-loaded with data by another source but need to
+     * be accessed by this file system. Strong existence is defined as the presence of a zero-length blob whose name is
+     * the directory path with a particular metadata field indicating the blob's status as a directory and is also known
+     * as a concrete directory. Directories created by this file system will strongly exist. Operations targeting
+     * directories themselves as the object (e.g. setting properties) will target marker blobs. Operations using the
+     * directory only as a container (e.g. listing) will operate on the blob-name prefix.
+     *
+     * This method fulfills the contract of: "The check for the existence of the file and the creation of the directory
```

Comment:
update: `This method fulfills the nio contract`... so we know what you're referencing.

Reply:
Additionally, this whole paragraph feels a little off. It reads like you're moving to another topic before I've understood the current one. Most notably, I don't understand how we are obeying the contract from this paragraph.